tests/serverpod_test_server/test_integration/logging/endpoint_method_stream_logging_test.dart

```diff
@@ -0,0 +1,375 @@
+  var client = Client('http://localhost:8080/');
+  late Serverpod server;
+  late Session session;
+
+  group('Database logger -', () {
+    setUp(() async {
+      server = IntegrationTestServer.create();
+      await server.start();
+
+      session = await server.createSession(enableLogging: false);
+      await LoggingUtil.clearAllLogs(session);
+    });
+
+    tearDown(() async {
+      await client.closeStreamingMethodConnections(exception: null);
+      client.close();
+      await await session.close();
+      await server.shutdown(exitProcess: false);
+    });
+
+    test(
+        'Given that continuous logging is turned on when sending a stream message without closing the connection then the log is created.',
+        () async {
+      var settings = RuntimeSettingsBuilder()
+          .withLogSettings(LogSettingsBuilder()
+              .withLogStreamingSessionsContinuously(true)
+              .build())
+          .build();
+      await server.updateRuntimeSettings(settings);
+
+      var controller = StreamController<int>();
+      var outputStream = client.logging.streamEmpty(controller.stream);
+
+      controller.add(1);
+      await awaitFirst(outputStream);
+
+      // Wait for the log to be written
+      await Future.delayed(Duration(milliseconds: 100));
+
+      var logs = await LoggingUtil.findAllLogs(session);
+
+      expect(logs, hasLength(1));
+      expect(logs.first.sessionLogEntry.isOpen, isTrue);
+    });
+
+    test(
+        'Given that continuous logging is turned on when sending a stream message and closing the connection then the log is created.',
+        () async {
+      var settings = RuntimeSettingsBuilder()
+          .withLogSettings(LogSettingsBuilder()
+              .withLogStreamingSessionsContinuously(true)
+              .build())
+          .build();
+      await server.updateRuntimeSettings(settings);
+
+      var controller = StreamController<int>();
+      var outputStream = client.logging.streamEmpty(controller.stream);
+
+      controller.add(1);
+      await outputStream.first;
+
+      // Wait for the log to be written
+      await Future.delayed(Duration(milliseconds: 100));
+
+      var logs = await LoggingUtil.findAllLogs(session);
+
+      expect(logs, hasLength(1));
+      expect(logs.first.sessionLogEntry.isOpen, isFalse);
+    });
+
+    test(
+        'Given that continuous logging is turned on when sending a stream message and writing a log without closing then the log is attached to the log session.',
+        () async {
+      var settings = RuntimeSettingsBuilder()
+          .withLogSettings(LogSettingsBuilder()
+              .withLogStreamingSessionsContinuously(true)
+              .build())
+          .build();
+      await server.updateRuntimeSettings(settings);
+
+      var controller = StreamController<int>();
+      var outputStream = await client.logging.streamLogging(controller.stream);
+
+      controller.add(1);
+      await outputStream.first;
```

Comment:
Question: The description here says without closing but it seems like the connection is closed because you only take first? Are we testing the correct thing here?

Reply:
first calls cancel on the stream automatically according to the documentation